### PR TITLE
6207 rapidpro outbound push

### PR DESCRIPTION
**Pull request by abbyad**
# Description

Adds support for using outbound push with RapidPro, or any third party tool that does authentication via credentials in HTTP headers.

Documentation and tests have yet to be included.

medic/cht-core#6207

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: sentinel/src/schedule/outbound.js [hunk 1]
```diff
@@ -159,6 +159,20 @@ const send = (payload, config) => {
         });
     }
 
+    if (authConf.type.toLowerCase() === 'header') {
+      if (authConf.name.toLowerCase() === 'authorization') {
```
Comment (SCdF) [inline sentinel/src/schedule/outbound.js]: If they don't provide an `authConf.name` this line will uncleanly crash the parsing.

Note on line 147 (and 143 before it) we deal with this for properties that are relevant generically. For you here, you can probably just change it to:

```suggestion
      if (authConf.name && authConf.name.toLowerCase() === 'authorization') {
```

Comment (SCdF) [reply sentinel/src/schedule/outbound.js]: Nit: I don't think there is that much value in having two if blocks nested inside one another here, you can just check both at the same time and save some precious whitespace.

**Comment by abbyad** [comment]
Added test and documentation, ready for another review.

File: sentinel/src/schedule/outbound.js (hunk 1 above)
Comment (abbyad) [reply sentinel/src/schedule/outbound.js]: Thanks, the intention was to keep clearly distinct blocks for each authConf type.

**Pull request merged**